src/object/cli_ec.c

```diff
@@ -365,11 +371,12 @@ ec_init_tgt_set(daos_iod_t *iods, unsigned int nr,
 			 */
 			D_ASSERT(!(PARITY_INDICATOR & rs));
 			for (ext_idx = rs; ext_idx <= re; ext_idx += len) {
-				unsigned int cell = (ext_idx % (k*len))/len;
+				unsigned int cell = (ext_idx % ss)/len;
 
 				*tgt_set |= 1UL << (cell+p);
 				if (*tgt_set == full) {
-					*tgt_set = 0;
+					if (parify_include)
```

Comment:
It's more compact to have:

if  (*tgt_set == full && parity_include) {
        *tgt_set = 0;
         return;
}

Reply:
right. thx